src/createRetryFetch.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+const defaultMaxRetry = 3;
+
+function defaultBackoffStrategy(retryCount) {
+  return new Promise((resolve) => setTimeout(resolve, retryCount * 100));
+}
+
+function createRetryFetch(enhancedFetch,
+  maxRetry = defaultMaxRetry,
+  backoffStrategy = defaultBackoffStrategy) {
+  let n = 0;
+  return (nextFetch) => (path, options = {}) => {
+    const retryFetch = () => enhancedFetch(nextFetch)(path, options)
```

Comment:
The pattern for these enhanced fetches is to wrap the call to the `nextFetch` argument:
```suggestion
function createRetryFetch(
  maxRetry = defaultMaxRetry,
  backoffStrategy = defaultBackoffStrategy
) {
  let n = 0;
  return (nextFetch) => (path, options = {}) => {
    const retryFetch = () => nextFetch(path, options)
```
then you can compose enhanced fetches:
```javascript
const enhanced = compose(
  createTimeoutFetch(6 * 1e3),
  createRetryFetch(3),
  createBrowserLikeFetch({ hostname, res })
)(fetch)
// or, manually
createTimeoutFetch(6 * 1e3)(
  createRetryFetch(3)(
    createBrowserLikeFetch({ hostname, res })(
      fetch
    )
  )
)
```
the order of that composition determines if the timeout applies to the retries or if retries should be attempted after timeout(s)

_or_ you could do both if you want to get fancy:
```javascript
// try 3 times to handle timeout errors
// try 2 times to handle network errors
// so worst-case total 6 retries
const enhanced = compose(
  createRetryFetch(3),
  createTimeoutFetch(6 * 1e3),
  createRetryFetch(2),
)(fetch)
```

Reply:
Love it. Updated!